libobs/util/platform-nix.c

```diff
@@ -752,10 +752,19 @@ os_inhibit_t *os_inhibit_sleep_create(const char *reason)
 
 #if defined(GIO_FOUND)
 	info->portal = portal_inhibit_info_create();
-	if (!info->portal)
+	if (!info->portal) {
+		/* In a Flatpak, only the portal can be used for inhibition */
+		if (access("/.flatpak-info", F_OK) == 0) {
+			bfree(info);
+			return NULL;
+		}
+
 		info->dbus = dbus_sleep_info_create();
-#endif
+	}
 
+	if (info->portal || info->dbus)
+		goto end;
```

Comment:
Rather than making this a `goto` and introducing the ifdef'd label at the end, can you not just make this an early return?

For example
```c
if (info->portal || info->dbus) {
	info->reason = bstrdup(reason);
	return info;
}
```

Reply:
Since we are in C, I tried to avoid duplicating code too much.

Reply:
Sure but by avoiding to duplicate 2 lines you've added three for the guarded goto label, which is pretty ugly. I'd prefer a simple early return as it also exists in the other case you've added. But Since this whole thing should probably be reworked at some point it's probably fine for now if we want to merge it for 30.2. I'll leave that up to @kkartaltepe

Reply:
personally I prefer just copying the code in this case, but you commented first.

Reply:
If you agree with Rodney, don't approve the PR without a comment indicating you would like the same change.

Removed all added `goto` with copied return.